src/main/java/org/prebid/server/analytics/AnalyticsReporterDelegator.java

```diff
@@ -72,4 +90,79 @@ public AnalyticsReporterDelegator(List<AnalyticsReporter> delegates,
                     tcfContext.getConsentString(), delegates);
         }
     }
+
+    private <T> void validateEvent(T event) {
+        if (event instanceof AuctionEvent) {
+            validateExtPrebidAnalytics((AuctionEvent) event);
+        }
+    }
+
+    private void validateExtPrebidAnalytics(AuctionEvent auctionEvent) {
+        final ExtRequest requestExt = auctionEvent.getAuctionContext().getBidRequest().getExt();
+        final ExtRequestPrebid extPrebid = requestExt != null ? requestExt.getPrebid() : null;
+        final ObjectNode analytics = extPrebid != null ? extPrebid.getAnalytics() : null;
+        final Iterator<String> analyticsFieldNames = analytics != null && !analytics.isEmpty()
+                ? analytics.fieldNames() : null;
+
+        if (analyticsFieldNames != null) {
+            final List<String> unknownAdapterNames = StreamSupport.stream(
+                    Spliterators.spliteratorUnknownSize(analyticsFieldNames, Spliterator.ORDERED), false)
+                    .filter(adapter -> !reporterNames.contains(adapter))
```

Comment:
You may use `anyMatch()` instead.

Reply:
But we use this one to get all unknown adapters and log them